fullcontact/api/base/base.py

```diff
@@ -0,0 +1,135 @@
+    DEFAULT_DELAY = 1.0
+    RETRY_STATUS_CODES = (429, 503)
+
+    def __init__(self,
+                 client_config: ClientConfig,
+                 headers: dict = None
+                 ):
+        r"""
+        Interface to communicate with FullContact API.
+
+        :param client_config: The config to be used to make
+        API calls.
+        :param headers: additional_headers to be passed. Authorization and Content-Type
+        are added automatically.
+        """
+        self.config = client_config
+        self._headers = self._get_validated_headers(headers)
+
+    @property
+    def _default_headers(self):
+        return {
+            "User-Agent": "FullContact_Python_Client_V%s" % __version__,
+            "Content-Type": "application/json",
+            "Authorization": "Bearer %s" % self.config.get_api_key()
+        }
+
+    @staticmethod
+    def _get_validated_headers(headers: dict) -> dict:
+        r"""
+        Check the type of the headers parameter.
+
+        :param headers: Headers dict to be validated.
+
+        :return: :param headers: if it is a dict with values. {} if None.
+
+        :raises: FullContactException if headers is not dict or None.
+        """
+        validated_headers = headers if headers is not None else {}
+        if type(validated_headers) != dict:
+            raise TypeError("Parameter headers should be of type 'dict'")
+        return validated_headers
+
+    def _get_merged_headers(self, *headers) -> dict:
+        r"""
+        Validate and merge header dictionaries.
+
+        :param headers: Multiple header dicts as args.
+        """
+        merged_headers = {}
+        for header in headers:
+            merged_headers.update(self._get_validated_headers(header))
+
+        return merged_headers
+
+    def _post_to_api(self,
+                     endpoint: str,
+                     data: dict = None,
+                     headers: dict = None
+                     ) -> Response:
+        r"""
+        Send POST request to API based on parameters.
+        Method type is always POST for the requests.
+
+        :param endpoint: endpoint to POST request to
+        :param data: body of the request to be sent
+        :param headers: additional_headers to be passed. Authorization and Content-Type
+        are added automatically.
+
+        :return: Response object
+        """
+        url = self._BASE_URL + endpoint
+        data = data or {}
+
+        final_headers = self._get_merged_headers(
+            self._headers,
+            headers,
+            self._default_headers)
+
+        return self.config.get_http_session().post(url=url,
+                                                   json=data,
+                                                   headers=final_headers)
+
+    def _validate_and_post_to_api(self,
+                                  request_validator: BaseSchema,
+                                  response_validator: Callable[[Response], BaseApiResponse],
```

Comment:
This is named `handler` or `validator` in a different place. These are more like request and response object. I would name it `api_request` and `api_response`. Each request/response object can have validate method or other methods.

Reply:
Got it. Change is on the way! 😄

Reply:
The reason I chose IdentityApi is that the endpoints are `identity.map`, `identity.resolve` and `identity.delete`. I wanted to try and make this similar to that. Currently, the methods that we have are `fullcontact_client.person.enrich`, `fullcontact_client.company.search`, etc. These resemble the API endpoints.
I can probable rename the class and keep the object names unchanged. Does that sound ok?